R/downloader.R

```diff
@@ -198,7 +191,36 @@ downloadWithRetries <- function(url, ..., maxTries = 5L) {
     )
     if (success) break
   }
-  success
+  if (!success) {
+    stop("Download failed.")
+  }
+  return(TRUE)
+}
+
+canUseRenvDownload <- function() {
+  getOption("packrat.authenticated.downloads.use.renv", FALSE) &&
+    identical(secureDownloadMethod(), "curl")
+}
+
+renvDownload <- function(url, destfile, method = inferAppropriateDownloadMethod(url), type = NULL, ...) {
+  debug <- getOption("packrat.renv.debug", FALSE)
+  if (debug) {
+    options(renv.download.trace = TRUE) # TODO remove
+  }
+
+  # We temporarily set our user agent to "curl" so that Bitbucket will treat us
+  # like a command line and not a browser. Otherwise, if we make unauthorized
+  # requests to Bitbucket .tar.gz URLs, we get redirects instead of a 401.
+  renv_useragent_option <- options("renv.http.useragent" = "curl")
```

Comment:
Should we only modify the user-agent when performing a BitBucket download with renv?